cogs/games.py

```diff
@@ -20,18 +20,203 @@
+            self.hangman_dict: Dict[str, Tuple[str, str]] = load(hangman_pkl)
+
+    @commands.max_concurrency(1, per=commands.BucketType.channel, wait=False)
+    @commands.command(aliases=["hm"])
+    async def hangman(self, ctx, *, command: str = None):
+        """Play a nice game of hangman with internet strangers!
+        Guesses must be single letters (guesses must be lower case letters to be valid) or the entire correct word
+		Can either be called with "?{hm|hangman}" or "?{hm|hangman} x", where x is a valid category command
+        Get all categories/help by typing "?{hm|hangman} help"
+        """
+        if command == "help":
+            await ctx.send(
+                f"rules: 6 wrong guesses are allowed, guesses must be either the entire correct word or a single lowercase letter\nhere is a list of valid category commands: {sorted(self.hangman_dict.keys())}"
+            )
+            return
+        if command is None:
+            category = choice(list(self.hangman_dict.keys()))
+        else:
+            category = command
+        try:
+            word_list, cat_name = self.hangman_dict[category]
+            word = choice(word_list).lower()
+        except KeyError:
+            await ctx.send(
+                f"invalid category command, here is a list of valid commands: {sorted(self.hangman_dict.keys())}"
+            )
+            return
+        num_mistakes = 0
+        not_guessed = set(re.sub(r"[^a-z]", "", word))
+        incorrect_guesses = set()
+        first_line = "".join(char + " " if char not in not_guessed else "_ "
+                             for char in word).rstrip()
+        last_line = "incorrect guesses: "
+        player_msg_list = []
+        timeout_dict = {}
+        invalid_msg_count = 0
+        same_channel_check = lambda msg: msg.channel == ctx.message.channel
+        txt_embed = Embed(colour=0xFF0000)
+        txt_embed.add_field(
+            name=f"hangman (category: {cat_name})",
+            value=
+            f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+        )
+        txt_embed.set_footer(text=last_line)
+        hg_msg = await ctx.send(embed=txt_embed)
+        while len(not_guessed) > 0 and num_mistakes < 6:
+            curr_msg = await self.bot.wait_for('message',
+                                               check=same_channel_check,
+                                               timeout=300)
+            curr_guess = curr_msg.content
+            if not (curr_msg.author in timeout_dict and
+                    (time() - timeout_dict[curr_msg.author]) < 3.0):
+                if curr_guess in "abcdefghijklmnopqrstuvwxyz" and len(
+                        curr_guess) == 1:
+                    await curr_msg.delete()
+                    if curr_guess in not_guessed:    # curr_guess in not_guessed => curr_guess is correct and new
+                        not_guessed.remove(curr_guess)
```

Comment:
this gets a bit uncomfortably indented, i wonder if there's a way with moving around the if/else cases and using `continue` or something to make it less nested

Reply:
i honestly dont know how, but ill try to think about it, see if i can come up with something